### ISSUE TITLE
DApp does not recognize funds in MetaMask wallet

### ISSUE THREAD

**Issue by rainerhoell** [Giveth/giveth-dapps-v2#408]
Hi there, I just tried to donate 60 xDAI to a project via Gnosis chain. I have 110 xDAI in my wallet. The app displays this correctly, so wallet and chain are correct, MetaMask is connected. When I click on "Donate" I get "You don't have enough funds.". See the two screenshots.

Address is rainerhoell.eth.
Browser is Brave Version 1.37.109 Chromium: 100.0.4896.60 (64-Bit).
OS is Win11.

Is this a bug in the new app or am I overlooking something? Or is this specific to Brave?

![Screenshot1](https://user-images.githubusercontent.com/101056184/161428627-a7020127-d55d-4018-828c-94cf259438ab.jpg)
![Screenshot2](https://user-images.githubusercontent.com/101056184/161428634-f4a32a2a-b90a-4f4e-a6e9-8782a73eba9a.jpg)

**Comment by MoeNick** [comment]
Are you still have this issue? @rainerhoell

**Comment by rainerhoell** [comment]
Yes, just tried again, same result.

**Comment by MoeNick** [comment]
Would you pls try to make your Brave Shield off, and let me know what will happen? @motoraif

**Comment by rainerhoell** [comment]
I just tried it with Brave Shield switched off. Same result, I'm afraid.

**Comment by rainerhoell** [comment]
And just tried the same under Firefox. Same error. So this is not a browser issue.
As you can see in Blockscout, the address used (rainerhoell.eth or 0xEeD8682Ea339Ed3E1966D7A557be082aD80bc1Ad) has a sufficient xDai balance: https://blockscout.com/xdai/mainnet/address/0xEeD8682Ea339Ed3E1966D7A557be082aD80bc1Ad/transactions
And as you can see in the screenshot above, the app also displays this xDai balance correctly.
Also, I can make transactions with the same wallet address and the same browser in other web3 apps. For example, I used a bridge to get the xDai from the Mainnet over to the Gnosis Chain.
So I really do not understand what's wrong here.

**Comment by rainerhoell** [comment]
Just tried the same with Meta Mask Mobile on iOS 15.4.1. Same story. So it‘s not even an OS issue.

**Comment by rainerhoell** [comment]
So I tried a few more things to find out what the problem is:
* I tried to do the same transaction with another wallet address. This would have worked (I cancelled the transaction).
* I tested the wallet address by swapping 10 xDai to GIV on Honeyswap. That worked.
* After that I tried again to use the address to donate to the Changemaker project. Error message again ("Insufficient funds.").
* After that I used the address to donate to a different project. Strangely, this now worked without an error message.
* Finally, I tried my original transaction again and now it worked!
I have no idea what kind of bug this is/was. Maybe something to do with handling a virgin address in connection with a project that had not received any donations yet.
